src/PacketPay.js

```diff
@@ -98,20 +99,39 @@ module.exports = ({
         transaction: BSVPayment.transaction
       })
       if (paymentResult.status !== 'success') {
-        throw new Error('Payment not processed')
+        return res
+          .status(402)
+          .set({
+            'x-bsv-payment-satoshis-paid': {
+              'reference': paymentResult.reference,
+              'envelope': BSVPayment.transaction
+            }
+          })
+          .json({
+            status: 'error',
+            code: 'ERR_PAYMENT_FAILED',
+            reference: paymentResult.reference,
+            description: 'Payment failed. A BSV payment is required to complete this request - Please resend payment.'
+          })
       }
     } catch (e) {
       return res.status(400).json({
         status: 'error',
         code: e.code || 'ERR_PAYMENT_FAILED',
         description: e.message
       })
-    }
+    }    
     req.packetpay = {
       satoshisPaid: requestPrice,
       reference: paymentResult.reference
     }
-    res.set('x-bsv-payment-reference', paymentResult.reference)
```

Comment:
The `x-bsv-payment-reference` should still be included in the response header per Ty's update.